### PR TITLE
update npm to latest version in Dockerfile

### PR DESCRIPTION
**Pull request by opadron**

### CODE REVIEW

**Comment by danlamanna** [comment]
This doesn't actually work for me:

````
Step 20/21 : RUN girder-install web --all-plugins
 ---> Running in 964c2c355146
module.js:471
    throw err;
    ^

Error: Cannot find module 'semver'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous>
(/usr/local/lib/node_modules/npm/lib/utils/unsupported.js:2:14)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
Traceback (most recent call last):
  File "/usr/local/bin/girder-install", line 11, in <module>
    load_entry_point('girder', 'console_scripts', 'girder-install')()
  File "/girder/girder/utility/install.py", line 362, in main
    parsed.func(parsed)
  File "/girder/girder/utility/install.py", line 214, in install_web
    plugins=opts.plugins, noPlugins=opts.no_plugins)
  File "/girder/girder/utility/install.py", line 181, in runWebBuild
    (' '.join(cmd), proc.returncode))
Exception: Web client install failed: `npm install --unsafe-perm
--no-save --production` returned 1.
The command '/bin/sh -c girder-install web --all-plugins' returned a
non-zero code: 1
````

Sidenote: It's worth mentioning that the newest version of the node
image we're inheriting from (node 8.2.1) comes with npm 5.3.0,
whenever we decide we should move to node 8.

On Thu, Jul 27, 2017 at 3:22 PM, Omar Padron <notifications@github.com> wrote:
> ________________________________
>
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/girder/girder/pull/2225
>
> Commit Summary
>
> update npm to latest version in Dockerfile
>
> File Changes
>
> M Dockerfile (1)
>
> Patch Links:
>
> https://github.com/girder/girder/pull/2225.patch
> https://github.com/girder/girder/pull/2225.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.



-- 
Dan LaManna
Kitware, Inc.

**Comment by brianhelba** [comment]
IMO, assuming that we're not forced to change or stay with our usage of Node.js v6 due to dependencies breaking, I think we should move to Nodejs v8 [in October 2017](https://github.com/nodejs/LTS).

Despite the painful churn, new versions of npm can and do bring us useful new features. However, Node.js is feature-complete in the way that we currently use it, so stability is paramount.

File: Dockerfile [hunk 1]
```diff
@@ -17,6 +17,7 @@ RUN apt-get update && apt-get install -qy software-properties-common python-soft
   apt-get clean && rm -rf /var/lib/apt/lists/*
 
 RUN wget https://bootstrap.pypa.io/get-pip.py && python get-pip.py
+RUN npm install -g npm
```
Comment (Xarthisius) [inline Dockerfile]: Don't do this like that. It needlessly increases size of the image, not to mention the fact that it gives you a "random" version of npm each time you build the image. If you want to change npm, update base image.

**Comment by zachmullen** [comment]
Let's close this and plan to update the base image in the near future.

**Comment by brianhelba** [comment]
I'd be ok if we use the `node:8.2` (but no earlier) base image (since that one uses npm 5.3) for the Docker container now, as long as we do a through test of the build with Node.js v8, which I have not yet done myself. We should still list Node.js v6 as the minimum / recommended version in our install docs, and run our CI tests with v6.

[In October](https://github.com/nodejs/LTS), we'll make Node.js v8 the absolute minimum for all Girder build types.

**Comment by brianhelba** [comment]
@zachmullen @Xarthisius @danlamanna @opadron See my last comment.

**Pull request closed without merging**